src/status_im/ui/screens/browser/events.cljs

```diff
@@ -58,6 +61,20 @@
      (-> (add-browser-fx cofx new-browser)
          (update-in [:db :browser/options] #(assoc % :browser-id (:browser-id new-browser)))))))
 
+(handlers/register-handler-fx
+ :update-browser-with-history
+ [re-frame/trim-v]
+ (fn [{:keys [now] :as cofx} [browser url]]
+   (let [history-index (if (nil? (get browser :history-index)) 0 (get browser :history-index))
+         history (get browser :history)
+         history-url (nth history history-index)
+         browser-history-to-index (apply list (drop history-index history))
+         new-history (if (not= history-url url) (conj browser-history-to-index url) history)
+         browser2 (-> browser (assoc-in [:history] new-history) (assoc-in [:history-index] 0))
```

Comment:
why not just `(assoc browser :history new-history :history-index 0)`

Reply:
Done!